### PR TITLE
Update data for the Popover API

### PR DESCRIPTION
**Pull request by chrisdavidmills**
#### Summary

The [Popover API](https://html.spec.whatwg.org/multipage/popover.html) is going to be available in Chrome 114. I've had this confirmed by the relevant engineers, and it is also reflected by the platform status entry: https://chromestatus.com/feature/5463833265045504.

Some of the compat data for this API was already added, but this PR completes it.

See my [research document](https://docs.google.com/document/d/1ryC_4hvkLom0E7aaNnAB82_Ki5Rb_cKgrOJGSBU9fuw/edit#) for further details of what has been added in this PR.

cc @keithamus

### CODE REVIEW

**Review by keithamus — APPROVED** [review]
LGTM. This might hit merge errors if https://github.com/mdn/browser-compat-data/pull/19358 is merged first.

**Comment by chrisdavidmills** [comment]
> LGTM. This might hit merge errors if #19358 is merged first.

Heh, fixed!

File: api/HTMLElement.json [hunk 1]
```diff
@@ -388,7 +388,8 @@
           "spec_url": "https://html.spec.whatwg.org/multipage/indices.html#event-beforetoggle",
           "support": {
             "chrome": {
-              "version_added": "114"
+              "version_added": "114",
+              "notes": "Only available on elements that include the <code>popover</code> attribute."
```
Comment (Elchi3) [inline api/HTMLElement.json:392]: Is this behavior per spec (then why do we have this note?) or is it a bug in Chrome? (should we add partial_implementation then?)

File: api/HTMLElement.json [hunk 2]
```diff
@@ -2092,7 +2093,8 @@
           "spec_url": "https://html.spec.whatwg.org/multipage/indices.html#event-toggle",
           "support": {
             "chrome": {
-              "version_added": "114"
+              "version_added": "114",
+              "notes": "Only available on non-<code>&lt;details&gt;</code> elements that include the <code>popover</code> attribute."
```
Comment (Elchi3) [inline api/HTMLElement.json:2097]: Same question

**Review by Elchi3 — COMMENTED** [review]
Looks good to me but I have a question.

**Review by Elchi3 — APPROVED** [review]
Great! 👍

**Pull request merged**